core/src/test/java/jenkins/widgets/HistoryPageFilterTest.java

```diff
@@ -293,6 +306,96 @@ public void test_laterItemsNotEvaluated() throws IOException {
         Assert.assertEquals(HistoryPageEntry.getEntryId(6), historyPageFilter.oldestOnPage);
     }
 
+    @Test
+    public void test_search_runs_by_build_number() throws IOException {
+        //given
+        HistoryPageFilter<ModelObject> historyPageFilter = newPage(5, null, null);
+        List<ModelObject> runs = newRuns(23, 24);
+        List<Queue.Item> queueItems = newQueueItems(25, 26);
+        //and
+        historyPageFilter.setSearchString("23");
+
+        //when
+        historyPageFilter.add(runs, queueItems);
+
+        //then
+        Assert.assertEquals(1, historyPageFilter.runs.size());
+        Assert.assertEquals(HistoryPageEntry.getEntryId(23), historyPageFilter.runs.get(0).getEntryId());
+    }
+
+    @Test
+    public void test_search_should_be_case_sensitive_for_anonymous_user() throws IOException {
+        //given
+        HistoryPageFilter<ModelObject> historyPageFilter = newPage(5, null, null);
+        //and
+        historyPageFilter.setSearchString("failure");
+        //and
+        List<ModelObject> runs = Lists.<ModelObject>newArrayList(new MockRun(2, Result.FAILURE), new MockRun(1, Result.SUCCESS));
+        List<Queue.Item> queueItems = newQueueItems(3, 4);
+
+        //when
+        historyPageFilter.add(runs, queueItems);
+
+        //then
+        Assert.assertEquals(0, historyPageFilter.runs.size());
+    }
+
+    @Test
+    @Ignore //User with insensitiveSearch enabled needs to be injected
+    public void test_search_runs_by_build_result() throws IOException {
+        //TODO: Set user with insensitiveSearch enabled
+        List<ModelObject> runs = ImmutableList.<ModelObject>of(new MockRun(2, Result.FAILURE), new MockRun(1, Result.SUCCESS));
+        assertOneMatchingBuildForGivenSearchStringAndRunItems("failure", runs);
+    }
+
+    @Test
+    @Ignore //User with insensitiveSearch enabled needs to be injected
```

Comment:
`Security380Test` does most of what you need. Just impersonate any user name instead of anon, `DummySecurityRealm` allows all user names. `User.get(…).addProperty(new UserSearchProperty(true))` should add the property to the user.